t/TestCore.pm

```diff
@@ -48,4 +51,47 @@ add_block_preprocessor(sub {
     }
 });
 
+# borrowed from Test::Nginx::Util
+sub gen_random_port (@) {
+    my (@unavailable_port_list) = @_;
+
+    push @unavailable_port_list, $ENV{TEST_NGINX_SERVER_PORT};
```

Comment:
This is wrong. The test block can also override the server port IIRC. You should use the `$ServerPort` variable provided by Test::Nginx::Util here instead.